prototype2/gdgem/NMXConfig.cpp

```diff
@@ -3,86 +3,89 @@
                   std::istreambuf_iterator<char>());
 
-  if (!reader.parse(str, root, 0)) {
+  try {
+    root = nlohmann::json::parse(jsonstring);
+  }
+  catch (...) {
     XTRACE(INIT, WAR, "Invalid Json file: %s", jsonfile.c_str());
     return;
   }
 
-  builder_type = root["builder_type"].asString();
+  builder_type = root["builder_type"].get<std::string>();
 
   if ((builder_type == "VMM2") || (builder_type == "VMM3")) {
     /**< \todo get from slow control? */
     auto tc = root["time_config"];
-    time_config.set_tac_slope(tc["tac_slope"].asInt());
-    time_config.set_bc_clock(tc["bc_clock"].asInt());
-    time_config.set_trigger_resolution_ns(tc["trigger_resolution"].asDouble());
-    time_config.set_target_resolution_ns(tc["target_resolution"].asDouble());
-    time_config.set_acquisition_window(tc["acquisition_window"].asUInt());
+    time_config.set_tac_slope(tc["tac_slope"].get<int>());
+    time_config.set_bc_clock(tc["bc_clock"].get<int>());
+    time_config.set_trigger_resolution_ns(tc["trigger_resolution"].get<double>());
+    time_config.set_target_resolution_ns(tc["target_resolution"].get<double>());
+    time_config.set_acquisition_window(tc["acquisition_window"].get<unsigned int>());
 
     auto sm = root["srs_mappings"];
     for (unsigned int index = 0; index < sm.size(); index++) {
-      auto fecID = sm[index]["fecID"].asInt();
-      auto vmmID = sm[index]["vmmID"].asInt();
-      auto planeID = sm[index]["planeID"].asInt();
-      auto strip_offset = sm[index]["strip_offset"].asInt();
+      auto fecID = sm[index]["fecID"].get<int>();
+      auto vmmID = sm[index]["vmmID"].get<int>();
+      auto planeID = sm[index]["planeID"].get<int>();
+      auto strip_offset = sm[index]["strip_offset"].get<int>();
       srs_mappings.set_mapping(fecID, vmmID, planeID, strip_offset);
     }
   }
 
   auto cx = root["clusterer x"];
-  clusterer_x.hit_adc_threshold = cx["hit_adc_threshold"].asUInt();
-  clusterer_x.max_strip_gap = cx["max_strip_gap"].asUInt();
-  clusterer_x.max_time_gap = cx["max_time_gap"].asDouble();
-  clusterer_x.min_cluster_size = cx["min_cluster_size"].asLargestUInt();
+  clusterer_x.hit_adc_threshold = cx["hit_adc_threshold"].get<unsigned int>();
+  clusterer_x.max_strip_gap = cx["max_strip_gap"].get<unsigned int>();
+  clusterer_x.max_time_gap = cx["max_time_gap"].get<double>();
+  clusterer_x.min_cluster_size = cx["min_cluster_size"].get<size_t>();
 
   auto cy = root["clusterer y"];
-  clusterer_y.hit_adc_threshold = cy["hit_adc_threshold"].asUInt();
-  clusterer_y.max_strip_gap = cy["max_strip_gap"].asUInt();
-  clusterer_y.max_time_gap = cy["max_time_gap"].asDouble();
-  clusterer_y.min_cluster_size = cy["min_cluster_size"].asLargestUInt();
+  clusterer_y.hit_adc_threshold = cy["hit_adc_threshold"].get<unsigned int>();
+  clusterer_y.max_strip_gap = cy["max_strip_gap"].get<unsigned int>();
+  clusterer_y.max_time_gap = cy["max_time_gap"].get<double>();
```

Comment:
LLVM coding standard for the whole file.

Reply:
This is an old file and renaming variables were not considered, only changing the json library.